name: Telegram Fact Generator

on:

  schedule:

    - cron: '*/30 * * * *'

jobs:

  fact_generator:

    runs-on: ubuntu-latest

    steps:

    - name: Checkout code

      uses: actions/checkout@v2

    - name: Set up Python

      uses: actions/setup-python@v2

      with:

        python-version: 3.8

    - name: Install dependencies

      run: |

        pip install requests

    - name: Generate fact and send to Telegram channel

      run: |

        import requests

        import openai

        import os

        def generate_fact():

            openai.api_key = os.environ['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']

            prompt = "Generate a unique fact"

            completions = openai.Completion.create(

                engine="text-davinci-003",

                prompt=prompt,

                max_tokens=1024,

                n=1,

                stop=None,

                temperature=0.5,

            )

            message = completions.choices[0].text

            return message

        def send_message(fact):

            url = "https://api.telegram.org/6115654835:AAE0loDn4S22m7MTadauS4tGbNSjOBU0S_E/sendMessage"

            data = {

                "chat_id": "@FactoriumAI",

                "text": fact,

            }

            response = requests.post(url, json=data)

            if response.status_code != 200:

                raise ValueError("Failed to send message: {}".format(response.text))

        fact = generate_fact()

        send_message(fact)


          
